000-mm2-connector-offsets-support.md

```diff
@@ -0,0 +1,449 @@
+* Monitoring the progress of running connectors on a per-source-partition basis 
+* Skipping records that cause issues with a connector and cannot be addressed using existing error-handling features
+
+## Proposal
+
+Strimzi should include mechanisms to let users manage offsets when utilising the KafkaMirrorMaker2 resources.
+Many users of MirrorMaker make use of this resource, rather than the KafkaConnector resource, so it's important to support it here as well.
+Similar, to [proposal #76][proposal-76], this proposal suggests introducing a new annotation on the KafkaMirrorMaker2 resource to trigger each operation.
+The operations are:
+* List offsets.
+* Alter offsets.
+* Reset offsets.
+
+The list offsets operation uses a ConfigMap for the output.
+The alter offsets operation uses a ConfigMap for the input.
+
+In addition, a separate new annotation is proposed to allow the user to select which MirrorMaker connector to apply the action to.
+
+### Request/Response from the API endpoints
+
+The response body received when using the GET endpoint, and the request body used when calling the PATCH endpoint have identical structures to allow users to easily fetch and then alter offsets.
+The structure of the body is not fully defined for generic source connectors.
+This is because source connectors determine their own structure for offsets depending on the source system they are integrating with.
+However, for each of the MirrorMaker connectors (MirrorSourceConnector, MirrorCheckpointConnector, MirrorHeartbeatConnector) we know the expected structure.
+
+MirrorSourceConnector format:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "cluster": "east-kafka",
+        "partition": 0,
+        "topic": "mirrormaker2-cluster-configs"
+      },
+      "offset": {
+        "offset": 0
+      }
+    }
+  ]
+}
+```
+
+MirrorCheckpointConnector format:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "partition": 4,
+        "topic": "inventory",
+        "group": "my-group"
+      },
+      "offset": {
+        "offset": 0
+      }
+    }
+  ]
+}
+```
+
+MirrorHeartbeatConnector format:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "sourceClusterAlias": "east-kafka",
+        "targetClusterAlias": "west-kafka"
+      },
+      "offset": {
+        "offset": 0
+      }
+    }
+  ]
+}
+```
+
+### New annotations
+
+All three actions will make use of a new annotation called `strimzi.io/connector-offsets`.
+The possible values will be `list`, `alter`, and `reset`.
+
+A second new annotation will be added called `strimzi.io/mirrormaker-connector`.
+It will be required when `strimzi.io/connector-offsets` is set.
+The value will be the name of the connector to apply the action to, for example `east-kafka->west-kafka.MirrorSourceConnector`.
+
+If the `strimzi.io/mirrormaker-connector` annotation is set, but the `strimzi.io/connector-offsets` annotation is missing the operator will take no action but add a warning message.
```

Comment:
what happens in the opposite scenario where the strimzi.io/connector-offsets annotation is set, but the strimzi.io/mirrormaker-connector annotation is missing? We say this later: 

_If only one of the `strimzi.io/connector-offsets` and `strimzi.io/mirrormaker-connector` annotations is present on the KafkaMirrorMaker2 the operator will leave the annotation on the resource and update the KafkaMirrorMaker2 CR status to include a warning message_

Maybe we don't need to mention anything here?